File: frontend/src/api/tenant.ts
import request from '@/utils/request'

export interface Tenant {
  id: number
  code: string
  name: string
  platform: string
  description: string
  logo: string
  status: number
  settings: Record<string, any>
  owner_id: number
  created_at: string
  updated_at: string
  role?: string // 用户在该租户中的角色
}

export interface CreateTenantParams {
  code: string
  name: string
  platform?: string
  description?: string
}

export interface UpdateTenantParams {
  name?: string
  description?: string
  logo?: string
  status?: number
  settings?: Record<string, any>
}

export interface AddUserParams {
  user_id: number
  role: 'owner' | 'admin' | 'member'
}

export interface SwitchTenantParams {
  tenant_id: number
}

export const tenantApi = {
  // 获取所有租户列表（管理员）
  list(params?: { page?: number; size?: number; platform?: string; status?: number }) {
    return request.get('/tenants', { params })
  },

  // 获取当前用户可访问的租户列表
  getMyTenants() {
    return request.get('/tenants/my')
  },

  // 获取租户详情
  get(id: number) {
    return request.get(`/tenants/${id}`)
  },

  // 创建租户
  create(data: CreateTenantParams) {
    return request.post('/tenants', data)
  },

  // 更新租户
  update(id: number, data: UpdateTenantParams) {
    return request.put(`/tenants/${id}`, data)
  },

  // 删除租户
  delete(id: number) {
    return request.delete(`/tenants/${id}`)
  },

  // 添加用户到租户
  addUser(tenantId: number, data: AddUserParams) {
    return request.post(`/tenants/${tenantId}/users`, data)
  },

  // 从租户移除用户
  removeUser(tenantId: number, userId: number) {
    return request.delete(`/tenants/${tenantId}/users/${userId}`)
  },

  // 更新用户角色
  updateUserRole(tenantId: number, userId: number, role: string) {
    return request.put(`/tenants/${tenantId}/users/${userId}/role`, { role })
  },

  // 切换当前租户
  switchTenant(data: SwitchTenantParams) {
    return request.post('/tenants/switch', data)
  }
}

File: backend/internal/models/tenant.go
package models

import (
	"time"

	"gorm.io/gorm"
)

// Tenant 租户/账套模型
type Tenant struct {
	ID          int64          `json:"id" gorm:"primaryKey"`
	Code        string         `json:"code" gorm:"uniqueIndex;size:50;not null"` // 租户编码，如: taobao_001
	Name        string         `json:"name" gorm:"size:100;not null"`             // 租户名称，如: 淘宝业务部
	Platform    string         `json:"platform" gorm:"size:20"`                   // 关联平台: taobao, douyin, kuaishou
	Description string         `json:"description" gorm:"size:500"`               // 描述
	Logo        string         `json:"logo" gorm:"size:500"`                      // Logo URL
	Status      int            `json:"status" gorm:"default:1"`                   // 状态: 1-启用, 0-禁用
	Settings    JSONB          `json:"settings" gorm:"type:jsonb"`                // 租户配置
	OwnerID     int64          `json:"owner_id"`                                  // 负责人ID
	CreatedAt   time.Time      `json:"created_at"`
	UpdatedAt   time.Time      `json:"updated_at"`
	DeletedAt   gorm.DeletedAt `json:"-" gorm:"index"`
}

// TableName 指定表名
func (Tenant) TableName() string {
	return "tenants"
}

// TenantStatus 租户状态
const (
	TenantStatusDisabled = 0
	TenantStatusEnabled  = 1
)

// TenantUser 租户用户关联（用户可以访问多个租户）
type TenantUser struct {
	ID        int64     `json:"id" gorm:"primaryKey"`
	TenantID  int64     `json:"tenant_id" gorm:"uniqueIndex:idx_tenant_user;not null"`
	UserID    int64     `json:"user_id" gorm:"uniqueIndex:idx_tenant_user;not null"`
	Role      string    `json:"role" gorm:"size:20;default:'member'"` // owner, admin, member
	CreatedAt time.Time `json:"created_at"`
}

// TableName 指定表名
func (TenantUser) TableName() string {
	return "tenant_users"
}

// TenantRole 租户角色
const (
	TenantRoleOwner  = "owner"  // 所有者
	TenantRoleAdmin  = "admin"  // 管理员
	TenantRoleMember = "member" // 成员
)

// CreateTenantRequest 创建租户请求
type CreateTenantRequest struct {
	Code        string `json:"code" binding:"required,min=2,max=50"`
	Name        string `json:"name" binding:"required,min=2,max=100"`
	Platform    string `json:"platform"`
	Description string `json:"description"`
}

// UpdateTenantRequest 更新租户请求
type UpdateTenantRequest struct {
	Name        string `json:"name"`
	Description string `json:"description"`
	Logo        string `json:"logo"`
	Status      *int   `json:"status"`
	Settings    JSONB  `json:"settings"`
}

// TenantWithStats 带统计的租户信息
type TenantWithStats struct {
	Tenant
	ShopCount  int64 `json:"shop_count"`
	OrderCount int64 `json:"order_count"`
	UserCount  int64 `json:"user_count"`
}
